clone/cmd/ctclone/ctclone.go

```diff
@@ -108,24 +108,24 @@ type ctFetcher struct {
 // Batch provides a mechanism to fetch a range of leaves.
 // Enough leaves are fetched to fully fill `leaves`, or an error is returned.
 // This implements batch.BatchFetch.
-func (cf ctFetcher) Batch(start uint64, leaves [][]byte) error {
+func (cf ctFetcher) Batch(start uint64, leaves [][]byte) (uint64, error) {
```

Comment:
@mhutchinson what's the reason the `Batch` function signature takes an "OUT" parameter rather than being something like `f(start, N uint64) ([][]byte, error) and just returning what was available`?

It feels like the `leaves` param is kinda doing a slightly unusual "double duty" here: informing the implementation of the max to fetch as well as being the container to put them in, which then leads to having to special case "short" reads.

Reply:
I can't remember if this was a performance thing, or trying to make this look something like `io.Reader`. I don't think the original contract was necessarily bad, nor is this change (which actually brings it closer to `io.Reader`, nor your suggestion.